src/Compilers/CSharp/Portable/BoundTree/OutDeconstructVarPendingInference.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class OutDeconstructVarPendingInference
+    {
+        public BoundDeconstructValuePlaceholder Placeholder;
+
+        public BoundDeconstructValuePlaceholder SetInferredType(TypeSymbol type, bool success)
```

Comment:
And we do have other examples of mutable bound nodes, UnboundLambda, for example. So, the principle is not that sacred, but we do prefer to follow it, unless there are good reasons not to.